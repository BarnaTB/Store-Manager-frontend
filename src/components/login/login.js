import React, { Component } from 'react';
import { connect } from 'react-redux';
import fetchLogin from '../../redux/actions/login/loginActions';
// import '../../css/signup.css';

export class Login extends Component {
    constructor(props){
        super(props)
        this.state = {

        }
    }
    onSubmit = (event) => {
        event.preventDefault()
        
        this.props.fetchLogin(this.state.username, this.state.password)

    };

    onUsernameChange = (event) => {
        this.setState({ [event.target.name]: event.target.value })
    };

    onPasswordChange = event => {
        this.setState({ [event.target.name]: event.target.value });
        
    };

    render() {
        return (
        <div className="">
            <a href="/"><h1>Store Manager</h1></a>
            <p>Welcome to your store Please log in!</p>
            <div className="">
                <form className="email-form" onSubmit={this.onSubmit}>
                    <input onChange={this.onUsernameChange} type="text" name='username' placeholder="Username . . ." id="username" required/><br /><br />
                    <input onChange={this.onPasswordChange} type="password" name='password' placeholder="Password . . ." id="password" required/><br /><br />
                    <input type="submit" value="Log in" id="submit-btn"/><br /><br />
                </form>
                <p>Don't have an account? <a href="/">Register</a></p>
            </div>
        </div>
        )
    }
}

export const mapStateToProps = (state) => {
    return state.loginReducer
}

export default connect(mapStateToProps, { fetchLogin })(Login);