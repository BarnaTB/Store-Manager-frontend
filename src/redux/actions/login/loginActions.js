import axios from 'axios';
import * as type from '../actionTypes';
import { toast } from 'react-toastify'

const fetchLogin = (username, password) => (dispatch) => {
    const data = { username, password }
    axios.post('https://store-manag.herokuapp.com/api/v1/login', data)
        .then((response) => {
            dispatch({
                type: type.LOGIN_SUCCESS,
                payload: response.data
            });
            toast.success(response.data.message)
            localStorage.setItem('token', response.data.token)
        })
        .catch((error) => {
            dispatch({
                type: type.LOGIN_FAILURE,
                payload: error
            });
            toast.error(error.response.data.message)
        });
    
};

export default fetchLogin;
