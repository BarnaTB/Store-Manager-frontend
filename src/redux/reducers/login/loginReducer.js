import { LOGIN_SUCCESS, LOGIN_FAILURE } from '../../actions/actionTypes';

const INITIAL_STATE = {
    success: false,
    payload: ''
}

const loginReducer = (state=INITIAL_STATE, action) => {
    switch (action.type) {
        case LOGIN_SUCCESS:
            return {
                ...state,
                success: true,
                payload: action.payload
            };
        case LOGIN_FAILURE:
            return {
                ...state,
                success: false,
                payload: action.payload
            };
        default:
            return state
    };
};

export default loginReducer;
