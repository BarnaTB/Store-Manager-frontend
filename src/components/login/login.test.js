import React from 'react';
import { shallow, mount } from 'enzyme';

import { Login } from './login';
import { mapStateToProps } from './login'

describe('App', () => {

    const mockfetchResetPasswordfn = jest.fn()

    const props ={
        fetchLogin: mockfetchResetPasswordfn,
        message: 'Logged in!',
        password:{value:''}
    }

    it('should render without crashing', () => {
        const wrapper = shallow(<Login />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should call onUsernameChange', () => {
        const onUsernameChangeMock = jest.fn();
        const event = {
        target: { name: 'username', value: 'Barnabas' }
        };
        const component = shallow(<Login onChange={onUsernameChangeMock} />);
        component.instance().onUsernameChange(event)
        expect(component.instance().state).toEqual({ username: 'Barnabas'});
    });

    test('should call onPasswordChange', () => {
        const onPasswordChangeMock = jest.fn();
        const event = {
        target: { name: 'password', value: 'Barna1234' }
        };
        const component = shallow(<Login onChange={onPasswordChangeMock} />);
        component.instance().onPasswordChange(event)
        expect(component.instance().state).toEqual({ password: 'Barna1234'});
    });

    test('should render a form', () => {
        const component = shallow(<Login/>)
        expect(component.find('form').length).toEqual(1);
    });

    test("should map states to props", () => {
        const mockedState = {
            loginReducer: {success: false, payload: ''}
        }
        const state = mapStateToProps(mockedState);
        expect(state).toEqual({success: false, payload: ''});
    });

    test('it should submit', () =>{
        const component = shallow(<Login {...props} />)
        component.setState({
            success: false,
            payload: ''
        })
        component.instance().onSubmit({preventDefault() {} })
        expect(component.state()).toEqual({"payload": "", "success": false});
    });
});
