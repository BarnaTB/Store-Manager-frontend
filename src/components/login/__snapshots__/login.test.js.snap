// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className=""
>
  <a
    href="/"
  >
    <h1>
      Store Manager
    </h1>
  </a>
  <p>
    Welcome to your store Please log in!
  </p>
  <div
    className=""
  >
    <form
      className="email-form"
      onSubmit={[Function]}
    >
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username . . ."
        required={true}
        type="text"
      />
      <br />
      <br />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password . . ."
        required={true}
        type="password"
      />
      <br />
      <br />
      <input
        id="submit-btn"
        type="submit"
        value="Log in"
      />
      <br />
      <br />
    </form>
    <p>
      Don't have an account? 
      <a
        href="/"
      >
        Register
      </a>
    </p>
  </div>
</div>
`;
